### PR TITLE
Hanythingondemand easyconfigs

### PR DESCRIPTION
**Pull request by ehiggs**
Included are easyconfigs for Hadoop, hanythingondemand, Spark, pbs_python, and an update to Python 2.7.9.

Enjoy!

### CODE REVIEW

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+name = 'Hadoop'
+version = '2.3.0-cdh5.0.0'
+
+homepage = 'http://archive.cloudera.com/cdh5/cdh/5/'
+description = """Hadoop MapReduce as provided by Cloudera"""
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+sources = ['%(namelower)s-%(version)s.tar.gz']
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb]: use `SOURCELOWER_TAR_GZ`

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+source_urls = ['http://archive.cloudera.com/cdh5/cdh/5/']
+
+java = 'Java'
+javaver = '1.7.0_40'
+
+dependencies = [(java, javaver), ('snappy', '1.2.2']
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb]: missing `)` at the end and split the deps up over multiple lines like:
```python
dependencies = [
    (java, javaver),
    ('snappy', '1.2.2'),
]
```

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.4.0-seagate.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+name = 'Hadoop'
+version = '2.4.0-seagate'
+
+homepage = 'https://github.com/Seagate/hadoop-on-lustre2'
+description = """Hadoop MapReduce for Parallel File Systems as provided by Seagate"""
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+sources = ['%(namelower)s-%(version)s.tar.gz']
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.4.0-seagate.eb]: `SOURCELOWER_TAR_GZ`

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.4.0-seagate.eb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+easyblock = 'Tarball'
+
+name = 'Hadoop'
+version = '2.4.0-seagate'
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.4.0-seagate.eb]: use `versionsuffix`?

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+easyblock = 'Tarball'
+
+name = 'Hadoop'
+version = '2.3.0-cdh5.0.0'
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb]: use `versionsuffix`?

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-2.1.4-intel-2015a-Python-2.7.9.eb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+installopts='--single-version-externally-managed --record=record'
+
+# a Python version with mpi4py and pbs_python (vsc.fancylogger) is required at runtime
+dependencies = [
+    (python, pythonver),
+    ('pbs_python', '4.3.5', versionsuffix),
+    ('vsc-mympirun', '3.4.2', '', True),
+]
+
+options = {'modulename': 'hod'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-2.1.4-intel-2015a-Python-2.7.9.eb]: `moduleclass`?

File: easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb [hunk 7]
```diff
@@ -0,0 +1,60 @@
+name = 'pbs_python'
+version = '4.3.5'
+easyblock = 'ConfigureMake'
+
+homepage = ''
+description = """The pbs_python package is a wrapper class for the Torque Batch System C library. With this package, you
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb]: watch the 120 column limit

File: easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+toolchainopts = {'usempi': True}
+
+source_urls = [
+    'http://ftp.sara.nl/pub/outgoing/',
+    'https://raw.github.com/adaptivecomputing/torque/4.2-dev/src/include/',
+    'https://oss.trac.surfsara.nl/pbs_python/raw-attachment/ticket/36/',
+    ]
+
+# torque-devel is missing a log.h file we need, so we ship it.
+my_startdir = "%s-%s/" % (name, version)
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb]: we've got `%(name)s` and `%(version)s`

File: easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+place_for_include_file = "include/torque"
+copy_cmd = 'mkdir -p %s%%s && cp %%%%s %s%%s ' % (my_startdir, my_startdir)
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb]: add newline below

File: easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+sources = [
+    SOURCE_TAR_GZ,
+    ('log.h', copy_cmd % (place_for_include_file, place_for_include_file)),
+    ('pbs_wrap_4.2.c', copy_cmd % ('src', 'src')),
+]
+patches = [
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/pbs_python/pbs_python-4.3.5-intel-2015a-Python-2.7.9.eb]: a newline above

File: easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb [hunk 11]
```diff
@@ -0,0 +1,19 @@
+easyblock = 'VersionIndependentPythonPackage'
+name = 'vsc-base'
+version = "1.9.1" # tweaked by EasyBuild (was: '1.7.3')
+
+homepage = 'http://hpcugent.github.com/vsc-mympirun/'
+description = """VSC-tools is a set of Python libraries and scripts that are commonly used within HPC-UGent."""
+
+# we build this to work with every python version
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
```
Comment (wpoely86) [inline easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb]: drop excess newline

File: easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb [hunk 12]
```diff
@@ -0,0 +1,19 @@
+easyblock = 'VersionIndependentPythonPackage'
+name = 'vsc-base'
```
Comment (wpoely86) [inline easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb]: newline above

File: easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+version = "1.9.1" # tweaked by EasyBuild (was: '1.7.3')
```
Comment (wpoely86) [inline easybuild/easyconfigs/v/vsc-base/vsc-base-1.9.1.eb]: drop `tweaked by`

File: easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    'files': ['bin/hadoop'],
+    'dirs': ['bin', 'lib', 'libexec']
+}
+
+modextrapaths = {
+    'HADOOP_HOME': 'share/hadoop/mapreduce',
+    'HADOOP_CONF_DIR': 'share/hadoop/mapreduce/conf'
+}
+
+postinstallcmds = ["cp $EBROOTSNAPPY/lib/libsnappy.so %(installdir)s/lib"]
```
Comment (boegel) [inline easybuild/easyconfigs/h/Hadoop/Hadoop-2.3.0-cdh5.0.0.eb]: where does this come from? no snappy dep?

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-2.1.4-intel-2015a-Python-2.7.9.eb (hunk 6 above)
Comment (boegel) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-2.1.4-intel-2015a-Python-2.7.9.eb]: no `modulename`, i.e. the name used for the `python -c 'import <modname>'` sanity check, this is correct

**Comment by ehiggs** [comment]
Closed in favour of #1418

**Comment by boegel** [comment]
@ehiggs: what about pbs_python, Spark?

I was actually working on an update of this, to clean it up to only include the bits that were not included in #1418

**Comment by ehiggs** [comment]
@boegel Could you submit that as a new PR then? ty

**Comment by boegel** [comment]
@ehiggs: OK, but we'll need to discuss which versions you need for `hanythingondemand` and deps.

The `pbs_python` easyconfig is a focking mess too, I'd like to discuss that with you first.

**Comment by boegel** [comment]
`hanythingondemand` and `pbs_python` easyconfigs in #1520, Hadoop in #1418

so let's close this as is?

**Pull request closed without merging**